File: android/src/main/kotlin/com/usercentrics/sdk/flutter/serializer/GppDataSerializer.kt
package com.usercentrics.sdk.flutter.serializer

import com.usercentrics.sdk.services.gpp.GppData
import com.usercentrics.sdk.services.gpp.GppSectionChangePayload

internal fun GppData.serialize(): Any {
    val serializedSections = sections.mapValues { (_, fields) ->
        fields.mapValues { (_, value) ->
            when (value) {
                null -> null
                is Boolean -> value
                is Int -> value
                is Double -> value
                is String -> value
                else -> value.toString()
            }
        }
    }
    return mapOf(
        "gppString" to gppString,
        "applicableSections" to applicableSections,
        "sections" to serializedSections
    )
}

internal fun GppSectionChangePayload.serialize(): Any {
    return mapOf(
        "data" to data
    )
}

File: ios/usercentrics_sdk/Sources/usercentrics_sdk/Serializer/GppDataSerializer.swift
import Usercentrics

extension GppData {
    func serialize() -> Any {
        return [
            "gppString": gppString,
            "applicableSections": applicableSections,
            "sections": sections,
        ]
    }
}

extension GppSectionChangePayload {
    func serialize() -> Any {
        return [
            "data": data,
        ]
    }
}
